accounts/accounts.go

```diff
@@ -46,41 +47,67 @@ func NewAccount(username string, hostname string) *Account {
 		Directory: directory,
 		Passwd:    "generated password",
 		NextId:    1,
+		Token:     token,
 	}
 }
 
-// TODO: Not threadsafe, or persistent
-func (account *Account) IncrementId() {
-	// TODO: Needs to be done correctly
-	account.NextId += 1
-}
+var current *Authentication
```

Comment:
Are there any trade-offs by making this global? Or is this just a convenience for the time being?

Reply:
I had started with encapsulating everything into an overall 'server' context, that was to be passed down to avoid having any globals. However, the handlers for net/http needed access, so I had switched to that horrible Global() call. But then, just to get it done, I basically made all of the parent data structs as Singletons (even though I really hate that pattern). So, really just convenience, and now somewhat inconsistent :-(